lms/static/scripts/frontend_apps/components/URLPicker.tsx

```diff
@@ -56,13 +56,38 @@ export default function URLPicker({
             : 'Annotating YouTube videos has been disabled at your organisation.',
         );
       } else {
-        onSelectURL(input.current!.value);
+        setError(null);
       }
     } catch (e) {
       setError('Please enter a URL, e.g. "https://www.example.com"');
     }
   };
 
+  // Sync HTML validation state with displayed error.
+  useLayoutEffect(() => {
+    input.current!.setCustomValidity(error ?? '');
+  }, [error]);
+
+  const onSubmit = (event: Event) => {
+    event.preventDefault();
+
+    const formEl = form.current!;
+    if (formEl.checkValidity()) {
+      onSelectURL(input.current!.value);
+    } else {
+      // Focus first field wth an error if invalid. This matches the behavior
+      // of native form validation, allowing the user to correct the error
+      // and also announcing the problem to screen reader users.
+      for (const el of Array.from(formEl.elements) as Array<
+        HTMLElement | HTMLInputElement
+      >) {
+        if ('validity' in el && !el.validity.valid) {
+          el.focus();
+        }
+      }
```

Comment:
`HTMLFormElement.reportValidity` does this normally as part of showing the browser-native validation error. Unfortunately it isn't possible to get the focusing without the native validation message.